packages/components/pie-icon-button/src/index.ts

```diff
@@ -1,20 +1,45 @@
-import { LitElement, html } from 'lit'; // eslint-disable-line import/no-extraneous-dependencies
-import { customElement } from 'lit/decorators.js'; // eslint-disable-line import/no-extraneous-dependencies
+import { LitElement, html, unsafeCSS } from 'lit'; // eslint-disable-line import/no-extraneous-dependencies
+import { customElement, property } from 'lit/decorators.js'; // eslint-disable-line import/no-extraneous-dependencies
+import { validPropertyValues } from '@justeattakeaway/pie-webc-core';
+import { ICON_BUTTON_VARIANT } from './defs';
+import styles from './iconButton.scss?inline';
 
-@customElement('pie-icon-button')
+// Valid values available to consumers
+export { ICON_BUTTON_VARIANT };
+
+const componentSelector = 'pie-icon-button';
+
+@customElement(componentSelector)
 export class PieIconButton extends LitElement {
-    // eslint-disable-next-line class-methods-use-this
+    @property()
+    @validPropertyValues(componentSelector, Object.values(ICON_BUTTON_VARIANT), ICON_BUTTON_VARIANT.PRIMARY)
+        variant : ICON_BUTTON_VARIANT = ICON_BUTTON_VARIANT.PRIMARY;
+
+    @property({ type: Boolean, reflect: true })
```

Comment:
Did we ever ask the LIT team about the reflect property and sb out of interest?

Reply:
Didn't ask them, but my understanding is that it means if the prop changes internally then the attribute in the DOM will update to reflect that value. It's probably not needed here.

Reply:
"You can configure a property so that whenever it changes, its value is reflected to its [corresponding attribute](https://lit.dev/docs/components/properties/#observed-attributes)."

https://lit.dev/docs/components/properties/#reflected-attributes

Reply:
removed as we don't need it :)